assets/js/components/Header.js

```diff
@@ -48,12 +49,16 @@ const Header = ( { children, subHeader, showNavigation } ) => {
 		select( CORE_USER ).isAuthenticated()
 	);
 	const { y } = useWindowScroll();
```

Comment:
@asvinb you removed this from the `EntityHeader` but is still used here. Is it still needed here?

Reply:
@aaemnnosttv Good question. Actually I think we can rely on the scroll event in `DashboardNavigation` to toggle the `googlesitekit-header--has-scrolled` class to the `Header` component and thus we no longer need to use the `useWindowScroll` hook. *However* I'am not too comfortable with this approach since `DashboardNavigation` is kinda updating the `Header` component whereas I would prefer to have each component changing its own markup/class names. What do you think about that?

Reply:
Thanks @asvinb – I agree, we shouldn't have components mutating the DOM rendered by another component.

I agree with @eugene-manuilov's idea above that we should be able to centralize this (or most of it) perhaps into a common parent? Thoughts @tofumatt ?

Reply:
100% agree we shouldn't be updating other component's DOM. 👍🏻 

Eugene's suggestion above https://github.com/google/site-kit-wp/pull/4621#discussion_r784785585 makes sense; I'm not sure if the hook is smart enough to combine listeners for performance, but if all we're doing is wanting to know if a scroll happened somewhere, either:

1. a custom hook (`useHasScrolled`)
2. Eugene's CSS approach

would be best. I prefer the CSS one as it's more flexible (it will apply to any children on the page)/less duplicated code.

Reply:
Thanks @tofumatt – `useWindowScroll` uses `useRafState` internally which does not share listeners from what I can see https://github.com/streamich/react-use/blob/3685b7502a4d1980953deed11c3dee31b220c89d/src/useRafState.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D

Reply:
@tofumatt @aaemnnosttv Just so that we're clear, even if we have the scroll logic to toggle a class name in the `Dashboard` component or `Root`, we will still need to have another scroll listener in `DashboardNavigation` for the chips to be active and also to know when the navigation becomes sticky. 
I can certainly see the benefits of moving the `--has-scrolled` class higher up the DOM since we can easily add any styles to any components.
Let me know if you're still on board with that approach.

Reply:
Fair enough, I guess the `DashboardNavigation` is a bit more specialised.

To me, that's fine, shame that `useWindowScroll` is creating multiple listeners but if we move the more generic "has scrolled" behaviour to `Root` then we at least won't make more for individual components in the future.

Let's still go with that approach, I'm on-board 🙂

Reply:
@aaemnnosttv After talking to @tofumatt we were planning to add the scroll logic in the `Root` component where the `googlesitekit-plugin--has-scrolled` class would have been added to `body`. However this would have resulted in unnecessarily adding scroll event listeners to pages where we won't ever use them, for e.g the settings page. Thus the best approach is going with the `useHasScrolled` hook where we just add the hook to the `DashboardMainApp` and `DashboardEntityApp` where we will be using them. Over for you to review 😁